en/salesforce/src/docs/TEMPLATE_ms_composer_custom_expression_editor.adoc

```diff
@@ -1,85 +1,304 @@
+
+|beginningOfYear |Returns a new DateTime value that changes the Month and Day values from the input to the first day of the specified year. It also sets the Time value to `00:00:00`. The month, day, and time in the input changes to `01-01T00:00:00`. |`beginningOfYear(2021-03-15T18:23:20) = 2021-01-01T00:00:00` +
+
+`beginningOfYear(now()) = 2021-01-01T00:00:00` |dateTime: DateTime |DateTime
+
+|daysBetween |Returns the number of days between two dates. If the to Date value is earlier than the from Date value, the function returns a negative number equal to the number of days between the two dates. |`daysBetween(2020-03-15, 2021-03-15) = 365`|from: Date +
+
+to: Date |Number
+
+|daysBetween |Returns the number of days between two DateTime values. If the to DateTime value is earlier than the from DateTime value, the function returns a negative number equal to the number of days between the two DateTime values. |`daysBetween(2020-03-15T18:23:20, 2021-03-15T02:23:20) = 365` |from: DateTime +
+
+to: DateTime |Number
+
+|month |Returns the numerical value of the month in the Date input. |`month(2021-03-15) = 3` |date: Date |Number
+
+|month |Returns the numerical value of the month in the DateTime input. |`month(2021-03-15T18:23:20) = 3` |dateTime: DateTime |Number
+
+|year |Returns the numerical value of the year in the Date input. |`year(2021-03-15) = 2021` |date: Date |Number
+
+|year |Returns the numerical value of the year in the DateTime input. |year(2021-03-15T18:23:20) = 2021 |dateTime: DateTime |Number
+
+|===
+
+== Data Types
+
+The value that a Composer field can accept for a value is a _data type_. In Composer, the following data types are available:
 
 [%header%autowidth.spread]
 
 |===
 
-|Composer Formula |DataWave Formula |Description
+|Data Type |Definition
+
+|Array | Provides a collection of a fixed number of components of the same data type.
+
+|Boolean | Evaluates to `true` or `false`.
+
+|Date | Stores a year, month, and day as `yyyy-MM-dd`.
 
-|today |Dates::today |Returns current date.
-|now |now | Returns current date and time.
-|addDays(#)
-|addMonths(#)
-|addYears(#)| Periods::days(#),
+|DateTime | Stores a date and time within a time zone relative to Greenwich Mean Time (GMT). A time zone must include either the symbol `+` in the time value or the symbol `-` in the date value. For example, 12:30 p.m. on June 1, 2019, in the Macau time zone is represented as `2019-06-01T12:30:00+08:00`.
+
+// |Integer | Stores whole numbers that range from -2,147,483,647 to 2,147,483,647. An integer has no decimal point.
+
+|Number | Stores either a floating-point number in the range of 1.40129846432481707e-45 to 3.40282346638528860e+38 or an integer.
+
+|Object | Refers to a composite data type that can contain other data types as properties  (`boolean`, `date`, `number`, and so on).
+
+|String | Stores text, numbers that are not `number` data types, or a combination of both, for example,  `94u0sfjlk*(f!`lkkjd`, `18005555555`, or `Fred`.
```

Comment:
```suggestion
|String | Stores text, numbers that are not `number` data types, or a combination of both, for example, `94u0sfjlk*(f!`lkkjd`, `18005555555`, or `Fred`.
```

Reply:
lol..."Fred"....

Reply:
I know! I wish I could find the OG doc I plagiarized this from. lol